start.sh

```diff
@@ -127,9 +127,22 @@ if [ -n "${PIHOLE_DNS_}" ]; then
           change_setting "PIHOLE_DNS_$count" "$i"
           ((count=count+1))
           ((valid_entries=valid_entries+1))
-        else
-          echo "Invalid IP detected in PIHOLE_DNS_: ${i}"
+          continue
         fi
+        if [ -n "$(dig +short $i)" ]; then
+          # If the "address" is a domain (for example a docker link) then try to resolve it and add 
+          # the result as a DNS server in setupVars.conf.
+          resolved_ip="$(dig +short $i | head -n 1)"
+          echo "Resolved ${i} from PIHOLE_DNS_ as: ${resolved_ip}"
+          if valid_ip "$resolved_ip" || valid_ip6 "$resolved_ip" ; then
+            change_setting "PIHOLE_DNS_$count" "$resolved_ip"
+            ((count=count+1))
+            ((valid_entries=valid_entries+1))
+            continue
+          fi
+        fi
+        # If the above tests fail then this is an invalid DNS server
+        echo "Invalid IP detected in PIHOLE_DNS_: ${i}"
```

Comment:
```suggestion
        echo "Invalid entry detected in PIHOLE_DNS_: ${i}"
```

Reply:
While you are under the hood changing this, please could you change it inside the `if [ $valid_entries -eq 0 ]` block? thanks!

Reply:
Fixed both in a new commit.